### PR TITLE
Updating Correlation Calculation (Making it more simpler and robust)

### PR DESCRIPTION
**Pull request by navdeep-G**
Modified docstring to show `uses` in h2o.var() and h2o.cor()
Modified calculation of Pearson Correlation Coefficient:
 - Removed CorTaskCompleteObs and now use CorTask as main worker in calculating correlation.
 - Removed CorTaskCompleteObsMean and now use CorMean to get mean across Vecs/columns.
 - Added another runit for h2o.cor(). Matches base R implementation. 
 - This new implementation is simpler and achieves the same results. 
 - Previously this had issues across multiple chunks. Does not seem to be the issue anymore.

---
This change is [Reviewable](https://reviewable.io/reviews/h2oai/h2o-3/87)

### CODE REVIEW

**Comment by abal5** [comment]
Can one of the admins verify this patch?

File: h2o-core/src/main/java/water/rapids/ast/prims/advmath/AstCorrelation.java [hunk 1]
```diff
@@ -91,9 +89,9 @@ private ValNum scalar(Frame frx, Frame fry, Mode mode) {
       if (!(Double.isNaN(xval) || Double.isNaN(yval)))
```
Comment (mmalohlava) [inline h2o-core/src/main/java/water/rapids/ast/prims/advmath/AstCorrelation.java]: Minor issue: use block statement - it is more readable

File: h2o-core/src/main/java/water/rapids/ast/prims/advmath/AstCorrelation.java [hunk 2]
```diff
@@ -88,12 +86,13 @@ private ValNum scalar(Frame frx, Frame fry, Mode mode) {
...
       xval = vecxs[r].at(0);
       yval = vecys[r].at(0);
-      if (!(Double.isNaN(xval) || Double.isNaN(yval)))
+      if (!(Double.isNaN(xval) || Double.isNaN(yval))) {
         //Compute variance of x and y vars
         xvar += Math.pow((vecxs[r].at(0) - xmean), 2);
-      yvar += Math.pow((vecys[r].at(0) - ymean), 2);
-      //Compute sum of squares of x and y
-      ss += (vecxs[r].at(0) - xmean) * (vecys[r].at(0) - ymean);
+        yvar += Math.pow((vecys[r].at(0) - ymean), 2);
```
Comment (mmalohlava) [inline h2o-core/src/main/java/water/rapids/ast/prims/advmath/AstCorrelation.java]: I am confused now, the `yvar` was before outside if statement, right now it is in `if` statement.

Comment (navdeep-G) [inline h2o-core/src/main/java/water/rapids/ast/prims/advmath/AstCorrelation.java]: It should be in block. Along with `xvar` and `ss`

Comment (navdeep-G) [inline h2o-core/src/main/java/water/rapids/ast/prims/advmath/AstCorrelation.java]: Not sure if it worked correctly before. New test catches all permutation of this method.

**Comment by navdeep-G** [comment]
I think its ready to merge if no other issues?

**Comment by mmalohlava** [comment]
One more thing before merging: i would squash the commits (at least the one with comments). And mark them with JIRA issues.

**Comment by navdeep-G** [comment]
Done.

**Pull request merged**